### ISSUE TITLE
Rule: dead templates

### ISSUE THREAD

**Issue by matthiasnoback** [matthiasnoback/phpstan-twig-analysis#2]
Not feasible (for now); see https://github.com/phpstan/phpstan/discussions/7227

- [x] We first switch to the new collector setup: collect template names from render calls, later analyze them all.
- [x] We should know about all the paths that may contain templates (maybe we can extract paths from the loader, or else let the user provide the template dir)
- [x] To save time, we can add an error about dead templates in the `CheckTwigRulesRule`.
- [x] Add back context for loader error: line and number where file was included/loaded, could be Twig file or PHP file
- [x] Add back context for Twig error: line and number from actual Twig file
- [x] Move template file finder to separate service, inject `TwigFactory`
- [x] Keep instantiated Twig Environment inside `TwigFactory` 
- [x] Extract service to extract template name from `MethodCall`.

**Comment by matthiasnoback** [comment]
At least this should give us a list of all the paths we should eventually scan for the template files that exist.

```php
<?php
declare(strict_types=1);

namespace PhpStanTwigAnalysis\Twig;

use Twig\Environment;
use Twig\Loader\ChainLoader;
use Twig\Loader\FilesystemLoader;
use Twig\Loader\LoaderInterface;

final class CollectTemplatePaths
{
    public function fromTwigEnvironment(Environment $twig): array
    {
        return $this->collectPathsFromLoader($twig->getLoader());
    }

    private function collectPathsFromLoader(LoaderInterface $loader): array
    {
        $paths = [];

        if ($loader instanceof ChainLoader) {
            foreach ($loader->getLoaders() as $subLoader) {
                $paths = array_merge($paths, $this->collectPathsFromLoader($subLoader));
            }
        } elseif ($loader instanceof FilesystemLoader) {
            foreach ($loader->getNamespaces() as $namespace) {
                foreach ($loader->getPaths($namespace) as $path) {
                    $paths[] = $path;
                }
            }
        }

        return $paths;
    }
}
```

**Comment by matthiasnoback** [comment]
Fixed by #27
